src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml.Serialization;
+using Dynamo.Models;
+using ProtoCore.Lang;
+
+namespace Dynamo.Core
+{
+    internal abstract class DynamoMigratorBase
+    {
+        protected readonly string rootFolder;
+
+        public string CurrentVersionPath
+        {
+            get
+            {
+                return Path.Combine(rootFolder, String.Format("{0}.{1}", MajorVersion, MinorVersion));
+            }
+        }
+
+        public abstract int MinorVersion { get; }
+        public abstract int MajorVersion { get; }
```

Comment:
Two things: `FileVersion` belongs to `DynamoMigrator.cs`, and it's a nice class. Then why don't we use `FileVersion` instead of having two `MinorVersion` and `MajorVersion` in `DynamoMigratorBase`? Just merge them into `FileVersion`, specifying one property will not leave the other out by mistake.